models/get-All-Articles.model.js

```diff
@@ -0,0 +1,42 @@
+const db = require("../db/connection")
+const { commentData } = require("../db/data/test-data")
+
+exports.selectAllArticles = () => {
+
+const comments = db.query(`SELECT *
+FROM articles
+JOIN comments
+ON articles.article_id = comments.article_id`)
+const articles = db.query(`SELECT * FROM articles ORDER BY created_at DESC`)
+
+return Promise.all([comments, articles])
+
+.then((array) => {
+
+const comments = array[0].rows
+const articles = array[1].rows
+
+   const minusBody = articles.map(article => {
+    delete article.body
+    article.commentCount = 0
+    return article})
+
+for(let i = 0 ; i < comments.length ; i++){
+let num = comments[i].article_id  
+   for(let j = 0 ; j < articles.length ; j++){
+    if(articles[j].article_id === num){
+        articles[j].commentCount = articles[j].commentCount + 1
+    }
+   }
+}
+    return minusBody
+})
+
```

Comment:
While this seems to work from what I can gather, it's quite difficult to read. I would have a look at Aggregate functions in the notes as this could be handled in the SQL call 😸 That being said, it doesn't mean that your way is wrong at all 😃 Changing this is not a requirement, but it is a recommendation 👍